common/tier4_autoware_utils/src/geometry/boost_polygon_utils.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tier4_autoware_utils/geometry/boost_geometry_utils.hpp"
+#include "tier4_autoware_utils/geometry/geometry.hpp"
+
+namespace
+{
+namespace bg = boost::geometry;
+using tier4_autoware_utils::Point2d;
+using tier4_autoware_utils::Polygon2d;
+
+void appendPointToPolygon(Polygon2d & polygon, const geometry_msgs::msg::Point & geom_point)
+{
+  Point2d point;
+  point.x() = geom_point.x;
+  point.y() = geom_point.y;
+
+  bg::append(polygon.outer(), point);
+}
+
+void appendPointToPolygon(Polygon2d & polygon, const Point2d point)
+{
+  bg::append(polygon.outer(), point);
+}
+
+double getPolygonArea(const geometry_msgs::msg::Polygon & footprint)
+{
+  double area = 0.0;
+
+  for (size_t i = 0; i < footprint.points.size(); ++i) {
+    size_t j = (i + 1) % footprint.points.size();
+    area += 0.5 * (footprint.points.at(i).x * footprint.points.at(j).y -
+                   footprint.points.at(j).x * footprint.points.at(i).y);
+  }
+
+  return area;
+}
+
+double getRectangleArea(const geometry_msgs::msg::Vector3 & dimensions)
+{
+  return static_cast<double>(dimensions.x * dimensions.y);
+}
+
+double getCircleArea(const geometry_msgs::msg::Vector3 & dimensions)
+{
+  return static_cast<double>((dimensions.x / 2.0) * (dimensions.x / 2.0) * M_PI);
+}
+}  // namespace
+
+namespace tier4_autoware_utils
+{
+bool isClockWise(const Polygon2d & polygon)
+{
+  const int n = polygon.outer().size();
+  const double x_offset = polygon.outer().at(0).x();
+  const double y_offset = polygon.outer().at(0).y();
+  double sum = 0.0;
+  for (std::size_t i = 0; i < polygon.outer().size(); ++i) {
+    sum +=
+      (polygon.outer().at(i).x() - x_offset) * (polygon.outer().at((i + 1) % n).y() - y_offset) -
+      (polygon.outer().at(i).y() - y_offset) * (polygon.outer().at((i + 1) % n).x() - x_offset);
+  }
+
+  return sum < 0.0;
+}
+
+Polygon2d inverseClockWise(const Polygon2d & polygon)
+{
+  Polygon2d inverted_polygon;
+  for (int i = polygon.outer().size() - 1; 0 <= i; --i) {
```

Comment:
pls check weather polygon.outer() is empty or not?

Reply:
also you can use std::reverse :+1:

Reply:
We do not have to check outer() is empty here. for loop is just ignored when outer() is empty.

Reply:
Thank you👍　How about to use std::reverse?

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/1367/commits/3931f55cc9311beeb5299b29da9d75533a46571c
DONE